doc/research/software-conflicts.md

```diff
@@ -0,0 +1,94 @@
+# Software conflicts in Agama
+
+Agama uses *yast2-packager* and *yast2-pkg-bindings* API for selecting patterns/packages and solving dependencies, calling to the *libzypp* C++ library under the hood.
+
+Sometimes there are conflicts with the selected packages/patterns. In that cases, the *zypp* resolver fails and reports a list of problems. In that cases, the YaST Packager UI shows the poblems to the user, allowing to select a solution for each problem.
+
+Agama needs something similar to YaST Packager in order to notify and fix the solver problems. This document analyzes how to bring that feature to Agama.
+
+## Solver problems from libzypp
+
+At low level, the solver problems are obtained in `libzypp` by calling to `#resolvePool` and then to `#problems` method:
+
+~~~
+bool success = zypp::getZYpp()->resolver()->resolvePool();
+zypp::ResolverProblemList = zypp::getZYpp()->resolver()->problems()
+~~~
+
+Each reported problem contains a list of possible solutions:
+
+~~~
+zypp::ResolverProblem
+    #details
+    #solutions (zypp::ProblemSolutionList)
+
+zypp::ProblemSolution
+    #description
+    #details
+~~~
+
+A list of solutions can be passed to `#applySolutions` method in order to fix the reported problems. The list of solutions is built by selecting a solution from each problem. Then the solver (`#resolvePool`) can be called again.
```

Comment:
I think `The list of solutions is built by selecting a solution from each problem` is not correct. As it does not need to be from each, but from any number of problems.